ethpm/validation.py

```diff
@@ -167,3 +174,18 @@ def validate_single_matching_uri(all_blockchain_uris: List[str], w3: Web3) -> st
             )
         )
     return matching_uris[0]
+
+
+def validate_uri_contents(contents: bytes, validation_hash: str) -> None:
+    """
+    Validate that the keccak(contents) matches the validation_hash.
+    """
+    hashed_contents = keccak(contents)
+    decoded_validation = decode_hex(validation_hash)
+    if hashed_contents != decoded_validation:
+        raise ValidationError(
+            "Invalid content-addressed URI. "
+            "Validation hash:{0} does not match the hash of URI contents: {1}.".format(
+                decoded_validation, hashed_contents
```

Comment:
Probably good to output these in their hex encoded formats for friendlier error messages.